File: shared/src/main/java/ru/privatenull/pnauth/extension/VerificationTicket.java
package ru.privatenull.pnauth.extension;
import java.time.Instant;
import java.util.UUID;
public record VerificationTicket(
        String id, String provider, UUID uniqueId, String username, AuthOperation operation,
        String message, Instant expiresAt, Status status
) {
    public enum Status { PENDING, APPROVED, DENIED, EXPIRED }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/extension/VerificationTicket.kt
package ru.privatenull.pnauth.extension

import java.time.Instant
import java.util.UUID

@JvmRecord
data class VerificationTicket(
    val id: String,
    val provider: String,
    val uniqueId: UUID?,
    val username: String?,
    val operation: AuthOperation,
    val message: String,
    val expiresAt: Instant,
    val status: Status
) {
    enum class Status { PENDING, APPROVED, DENIED, EXPIRED }
}
